modules/report/includes/commerce_pos_report.pages.inc

```diff
@@ -1005,3 +1004,300 @@ function commerce_pos_report_build_journal_role_table(array $filters = array())
+  global $user;
+  $form['#tree'] = TRUE;
+  $form['header'] = array(
+    '#markup' => theme('commerce_pos_header', array('account' => $user)),
+  );
+
+  $query_filter_defaults = array(
+    'results_per_page' => COMMERCE_POS_REPORT_KPI_REPORT_DEFAULT_ITEMS_PER_PAGE,
+  );
+
+  $query_params = drupal_get_query_parameters() + $query_filter_defaults;
+
+  $form['filters'] = array(
+    '#type' => 'container',
+  );
+
+  if (!isset($form_state['results_container_id'])) {
+    $form_state['results_container_id'] = 'commerce-pos-report-journal-role-container';
+  }
+
+  $js_settings = array(
+    'commercePosReport' => array(
+      'cssUrl' => url(drupal_get_path('module', 'commerce_pos_report') . '/css/commerce_pos_report_receipt.css', array(
+        'absolute' => TRUE,
+      )),
+    ),
+  );
+
+  $form['#attached']['libraries_load'][] = array('jquery-print');
+  $form['#attached']['css'][] = drupal_get_path('module', 'commerce_pos') . '/css/commerce_pos_style.css';
+  $form['#attached']['css'][] = drupal_get_path('module', 'commerce_pos_report') . '/css/commerce_pos_report.css';
+  $form['#attached']['js'][] = drupal_get_path('module', 'commerce_pos_report') . '/js/commerce_pos_report.journal_role.js';
+  $form['#attached']['js'][] = array(
+    'type' => 'setting',
+    'data' => $js_settings,
+  );
+  $form['report-options'] = array(
+    '#theme' => 'commerce_pos_report_options',
+  );
+  $form['options'] = array(
+    '#type' => 'container',
+    '#id' => 'commerce-pos-report-journal-role-options-container',
+  );
+  $form['options']['results_per_page'] = array(
+    '#type' => 'select',
+    '#title' => t('Items per page'),
+    '#options' => array(
+      25 => '25',
+      50 => '50',
+      100 => '100',
+      500 => '500',
+      -1 => t('all'),
+    ),
+    '#default_value' => $query_params['results_per_page'],
+    '#attributes' => array(
+      'class' => array('commerce-pos-report-journal-role-filter'),
+    ),
+  );
+  $form['results'] = array(
+    '#type' => 'container',
+    '#id' => $form_state['results_container_id'],
+  );
+  $form['results']['table'] = commerce_pos_report_build_kpi_report_table($query_params);
+  // Pager has to come after we've build the journal role table.
+  $form['options']['pager'] = array('#theme' => 'pager');
+  $form['results']['pager'] = array('#theme' => 'pager');
+
+  $form['submit'] = array(
+    '#value' => t('Submit'),
+    '#type' => 'submit',
+    '#weight' => -50,
+    '#attributes' => array(
+      'class' => array('element-invisible', 'commerce-pos-report-journal-role-submit'),
+    ),
+  );
+
+  return $form;
+}
+
+/**
+ * Submit handler for the journal role report form.
+ */
+function commerce_pos_report_kpi_report_submit($form, &$form_state) {
+  $query_params = array_merge(drupal_get_query_parameters(), $form_state['values']['options']);
+  $form_state['redirect'] = array(current_path(), array('query' => $query_params));
```

Comment:
I do have a concern on this , The old code is following the same ideology so i believe it would be better if we keep it same ? Let me know your thoughts?